StripeFinancialConnections/StripeFinancialConnections/Source/Helpers/FinancialConnectionsFont.swift

```diff
@@ -0,0 +1,106 @@
+            font = UIFont.systemFont(ofSize: 24, weight: .bold)
+            lineHeight = 32
+            appleTextStyle = .title2
+        }
+        return .create(font: font, lineHeight: lineHeight, appleTextStyle: appleTextStyle)
+    }
+
+    enum BodyToken {
+        case small
+        case smallEmphasized
+        case medium
+        case mediumEmphasized
+    }
+    static func body(_ token: BodyToken) -> FinancialConnectionsFont {
+        let font: UIFont
+        let lineHeight: CGFloat
+        let appleTextStyle: UIFont.TextStyle
+        switch token {
+        case .small:
+            // 14 size / 20 line height / 400 weight
+            font = UIFont.systemFont(ofSize: 14, weight: .regular)
+            lineHeight = 20
+            appleTextStyle = .footnote
+        case .smallEmphasized:
+            // 14 size / 20 line height / 600 weight
+            font = UIFont.systemFont(ofSize: 14, weight: .semibold)
+            lineHeight = 20
+            appleTextStyle = .footnote
+        case .medium:
+            // 16 size / 24 line height / 400 weight
+            font = UIFont.systemFont(ofSize: 16, weight: .regular)
+            lineHeight = 24
+            appleTextStyle = .body
+        case .mediumEmphasized:
+            // 16 size / 24 line height / 600 weight
+            font = UIFont.systemFont(ofSize: 16, weight: .semibold)
+            lineHeight = 24
+            appleTextStyle = .body
+        }
+        return .create(font: font, lineHeight: lineHeight, appleTextStyle: appleTextStyle)
+    }
+
+    enum LabelToken {
+        case largeEmphasized
+    }
+    static func label(_ token: LabelToken) -> FinancialConnectionsFont {
+        let font: UIFont
+        let lineHeight: CGFloat
+        let appleTextStyle: UIFont.TextStyle
+        switch token {
+        case .largeEmphasized:
+            // 16 size / 24 line height / 600 weight
+            font = UIFont.systemFont(ofSize: 16, weight: .semibold)
+            lineHeight = 24
+            appleTextStyle = .body
+        }
+        return .create(font: font, lineHeight: lineHeight, appleTextStyle: appleTextStyle)
+    }
+
+    private static func create(font: UIFont, lineHeight: CGFloat, appleTextStyle: UIFont.TextStyle) -> FinancialConnectionsFont {
+        let scaledFont = scaleFont(font, appleTextStyle: appleTextStyle)
+        return FinancialConnectionsFont(
+            uiFont: scaledFont,
+            lineHeight: scaleLineHeight(lineHeight, font: font, scaledFont: scaledFont)
+        )
+    }
+
+    private static func scaleFont(_ font: UIFont, appleTextStyle: UIFont.TextStyle) -> UIFont {
+        let metrics = UIFontMetrics(forTextStyle: appleTextStyle)
+        let scaledFont = metrics.scaledFont(for: font)
+        return scaledFont
+    }
+
+    private static func scaleLineHeight(_ lineHeight: CGFloat, font: UIFont, scaledFont: UIFont) -> CGFloat {
+        return lineHeight * (scaledFont.pointSize / font.pointSize)
```

Comment:
Are we guaranteed to denominator will never be 0 here?

Reply:
Probably unlikely but worth adding extra safety, thank you!